src/main/java/org/prebid/server/bidder/advangelists/AdvangelistsBidder.java

```diff
@@ -0,0 +1,261 @@
+import com.iab.openrtb.request.App;
+import com.iab.openrtb.request.Banner;
+import com.iab.openrtb.request.BidRequest;
+import com.iab.openrtb.request.Format;
+import com.iab.openrtb.request.Imp;
+import com.iab.openrtb.request.Site;
+import com.iab.openrtb.request.Video;
+import com.iab.openrtb.response.BidResponse;
+import com.iab.openrtb.response.SeatBid;
+import io.vertx.core.MultiMap;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.json.DecodeException;
+import io.vertx.core.json.Json;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.prebid.server.bidder.Bidder;
+import org.prebid.server.bidder.model.BidderBid;
+import org.prebid.server.bidder.model.BidderError;
+import org.prebid.server.bidder.model.HttpCall;
+import org.prebid.server.bidder.model.HttpRequest;
+import org.prebid.server.bidder.model.Result;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.proto.openrtb.ext.ExtPrebid;
+import org.prebid.server.proto.openrtb.ext.request.advangelists.ExtImpAdvangelists;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class AdvangelistsBidder implements Bidder<BidRequest> {
+    private static final TypeReference<ExtPrebid<?, ExtImpAdvangelists>> ADVANGELISTS_EXT_TYPE_REFERENCE = new
+            TypeReference<ExtPrebid<?, ExtImpAdvangelists>>() {
+            };
+    private static final String DEFAULT_BID_CURRENCY = "USD";
+    private final String endpointUrl;
+
+    public AdvangelistsBidder(String endpointUrl) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final List<BidderError> errors = new ArrayList<>();
+        List<HttpRequest<BidRequest>> httpRequests = new ArrayList<>();
+        try {
+            final Map<ExtImpAdvangelists, List<Imp>> impToExtImp = getImpToExtImp(request, errors);
+            httpRequests.addAll(buildAdapterRequests(request, impToExtImp));
+        } catch (PreBidException e) {
+            return Result.of(Collections.emptyList(), errors);
+        }
+
+        return Result.of(httpRequests, errors);
+    }
+
+    private static Map<ExtImpAdvangelists, List<Imp>> getImpToExtImp(BidRequest request, List<BidderError> errors) {
+        final List<Imp> imps = new ArrayList<>();
+        final List<ExtImpAdvangelists> extImpList = new ArrayList<>();
+        for (Imp imp : request.getImp()) {
+            try {
+                final ExtImpAdvangelists extImpEmxDigital = getExtImpAdvangelists(imp);
+                validateImpExt(extImpEmxDigital);
```

Comment:
You can do a validation in the same method where you parse the extension - just rename it to `parseAndValidateImpExt`